src/modules/dashboard/transformers.ts

```diff
@@ -157,3 +163,57 @@ export const getActionsListData = (
     },
   );
 };
+
+export const getEventsListData = (
+  unformattedEvents?: SubgraphEvents,
+): FormattedEvent[] | undefined =>
+  unformattedEvents?.events?.reduce((processedEvents, event) => {
+    if (event) {
+      const {
+        id,
+        associatedColony: { colonyAddress },
+        transaction: {
+          hash,
+          block: { timestamp },
+        },
+        name,
+        args,
+      } = event;
+      const {
+        agent,
+        domainId,
+        recipient,
+        fundingPotId,
+        metadata,
+        token,
+        paymentId,
+        amount,
+        payoutRemainder,
+        decimals = '18',
+      } = JSON.parse(args);
```

Comment:
it can throw error

Reply:
Yes and no.

`args` will always exist if we have an event. And even if it's an empty string: `''`, it will still be parsed by `JSON.parse`.

But yeah, I'll put in a provision to fallback to a valid value if it does not exist